netutils/netinit/netinit.c

```diff
@@ -411,45 +508,26 @@ static void netinit_set_ipaddrs(void)
 #endif
 
 /****************************************************************************
- * Name: netinit_net_bringup()
+ * Name: netinit_dhcp()
  *
  * Description:
- *   Bring up the configured network
+ *   Renew or obtain a IP Address over DHCP
  *
  ****************************************************************************/
 
-#if defined(NETINIT_HAVE_NETDEV) && !defined(CONFIG_NETINIT_NETLOCAL)
-static void netinit_net_bringup(void)
-{
 #ifdef CONFIG_NETINIT_DHCPC
+static int netinit_dhcp(struct dhcpc_state *ds)
+{
   uint8_t mac[IFHWADDRLEN];
-  struct dhcpc_state ds;
   FAR void *handle;
-#endif
-
-  /* Bring the network up. */
-
-  if (netlib_ifup(NET_DEVNAME) < 0)
-    {
-      return;
-    }
-
-#ifdef CONFIG_WIRELESS_WAPI
-  /* Associate the wlan with an access point. */
-
-  if (netinit_associate(NET_DEVNAME) < 0)
+  int ret = -EAGAIN;
+#ifdef CONFIG_FSUTILS_IPCFG
+  if ((g_netconf.proto & IPv4PROTO_DHCP) == 0)
```

Comment:
Hmmm... & makes not sense here.  The prototype is not bit encoded, it is an enumeration.

Reply:
Because of the particular definitions, that does test work.  But that is merely a quirk of fate, not by design.  By designed it was just an enumerated value.

Reply:
That was by design. I the original design, if you recall , it was called it flags. and I purposely spaced the code points. Were I obviously failed was documenting the bit of cleverness.  What was merely a quirk of fate, is that in the transitions from the boardctl to the ipcfg that  code point spacing was preserved.  did check it.